### ISSUE TITLE
Can't parse demos

### ISSUE THREAD

**Issue by 1mamute** [n1c/csgo-dem2json#2]
Hello dude,

Very interested in this idea.
If it does what is promised keep all relevant information after parsing to JSON (that most problably will be minified and compressed on my end ahaha) this is good news.

Just tried the solution here, the steps:
>Cloned the repo,
>npm run compile didnt worked so I
>npm i (added 143 packages from 130 contributors and audited 144 packages in 21.286s)
>npm run parse test.dem ~/output.json

And got this error
> csgo-dem2json@0.0.1 parse D:\Projetos\csgo-dem2json
> node dist/index.js "test.dem" "~/output.json"

```
Processing test.dem to ~/output.json
cs_match_end_restart { name: 'cs_match_end_restart', event: {} }
events.js:291
      throw er; // Unhandled 'error' event
      ^

RangeError: Illegal length: 0 <= 361167691 + 152 <= 361167824
    at ByteBuffer.module.exports.ByteBufferPrototype.skip (D:\Projetos\csgo-dem2json\node_modules\bytebuffer\dist\bytebuffer-node.js:2740:23)    
    at DemoFile._handleDemoPacket (D:\Projetos\csgo-dem2json\node_modules\demofile\dist\demo.js:168:19)
    at DemoFile._parseRecurse (D:\Projetos\csgo-dem2json\node_modules\demofile\dist\demo.js:240:16)
    at processImmediate (internal/timers.js:461:21)
Emitted 'error' event on DemoFile instance at:
    at DemoFile._parseRecurse (D:\Projetos\csgo-dem2json\node_modules\demofile\dist\demo.js:269:12)
    at processImmediate (internal/timers.js:461:21)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! csgo-dem2json@0.0.1 parse: `node dist/index.js "test.dem" "~/output.json"
```

If you have any ideia why I cant parse the demos it would be appreciated.
Hope to contribute in this project in the close future!
Peace

**Comment by n1c** [comment]
Hey bud! That error looks really weird, any chance you could share the demo? And have you tried with any others?

**Comment by 1mamute** [comment]
I did a clean reinstall just in case

```
cloned the repo
npm i
npm run compile
npm run parse ../test.dem ../output.json
```

Same error

The demo Im trying to parse is this: 
[Game 1 - Mirage - Vitality vs FaZe IEM NY 2020](https://www.hltv.org/matches/2344379/vitality-vs-faze-iem-new-york-2020-europe)

>  And have you tried with any others?

I just tried another demo but this time I downloaded directly from the CSGO client and then moved the .dem file from _Steam\steamapps\common\Counter-Strike Global Offensive\csgo\replays_ 

After that I tried downloading the exact same demo from HLTV.org
[Game 2 - Inferno - ENCE vs Astralis IEM Katowice 2019](https://www.hltv.org/matches/2331081/ence-vs-astralis-iem-katowice-2019)

Both went nicely. From 250mb .dem file to a 12mb .json. This is great for my use!

At first I thought HLTV.org did something to the files when uploading the demos, but it worked with this older demo downloading both from Valve servers and HLTV.org. Any chance that new demos have some new protobuf informations that is messing the parser?

I also tried parsing the first demo with saul/demofile, the same buffer error ocurred, weird indeed!
